### PR TITLE
MIM-121: Get offers by listing id

### PR DESCRIPTION
**Pull request by momentiris**

### CODE REVIEW

File: src/services/lease-service/tests/adapters/offer-adapter.test.ts [hunk 1]
```diff
@@ -276,4 +263,64 @@ describe('offer-adapter', () => {
...
+            listingId: listing.data.id,
+            offeredApplicant: expect.objectContaining({ id: applicants[1].id }),
+            selectedApplicants: [
+              expect.objectContaining({ id: applicants[1].id }),
+            ],
+          }),
+        ])
+      )
+    })
+  })
```
Comment (driatic) [inline src/services/lease-service/tests/adapters/offer-adapter.test.ts:325]: Add a test for the error handling in the adapter?

File: src/services/lease-service/tests/routes/offers.test.ts [hunk 2]
```diff
@@ -273,4 +273,32 @@ describe('offers', () => {
...
+
+      expect(res.status).toBe(200)
+      expect(res.body).toMatchObject({
+        content: expect.arrayContaining([
+          expect.objectContaining({ id: expect.any(Number) }),
+        ]),
+      })
+    })
+
+    it('should return an error if not found', async () => {
```
Comment (driatic) [inline src/services/lease-service/tests/routes/offers.test.ts:293]: Maybe the test name is what confuses me here, but I would expect an empty list and not error if there are no offers for a listing? 

Maybe rename the test to "should handle errors" or something like that?

**Review by driatic — APPROVED** [review]
Looks great! Two suggestions

File: src/services/lease-service/tests/adapters/offer-adapter.test.ts (hunk 1 above)
Comment (momentiris) [reply src/services/lease-service/tests/adapters/offer-adapter.test.ts:334]: https://github.com/Bostads-AB-Mimer/onecore-leasing/pull/131/commits/b609238cf8b985dc807afcbab2519aab19100bd9

File: src/services/lease-service/tests/routes/offers.test.ts (hunk 2 above)
Comment (momentiris) [reply src/services/lease-service/tests/routes/offers.test.ts]: https://github.com/Bostads-AB-Mimer/onecore-leasing/pull/131/commits/647e9b308267405bd8d6fc09fa3b88a1d834f443

Yeah it was confusing 👍

**Pull request merged**